SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -235,5 +248,101 @@ internal class InAppReceipt {
             }
             task.resume()
     }
+  
+    /**
+     *  Verify the purchase of a Consumable, NonConsumable, FreeSubscription or NonRenewingSubscription product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter inReceipt: the receipt to test in
+     */
+    class func verifyPurchase(
+        productId productId: String,
+        inReceipt receipt: ReceiptInfo
+    ) -> SwiftyStoreKit.VerifyPurchaseResult {
+      
+        // Get receipts info for the product
+        let receiptsInfo = getReceiptsInfo(forProductId: productId, inReceipt: receipt)
+      
+        // Verify that at least one receipt has the right product id
+        return receiptsInfo.count >= 1 ? .Purchased : .NotPurchased
+    }
+  
+    /**
+     *  Verify the purchase of a AutomaticallyRenewableSubscription product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter inReceipt: the receipt to test in
+     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
+     *  - Parameter validDuration: the duration of the subscription. Only required for non-renewable subscription.
+     */
+    class func verifySubscription(
+        productId productId: String,
+        inReceipt receipt: ReceiptInfo,
+        validUntil date: NSDate = NSDate(),
+        validDuration duration: NSTimeInterval? = nil
+    ) -> SwiftyStoreKit.verifySubscriptionResult {
+      
+        // Verify that at least one receipt has the right product id
+        let receiptsInfo = getReceiptsInfo(forProductId: productId, inReceipt: receipt)
+        if receiptsInfo.count == 0 {
+            return .NotPurchased
+        }
     
+        // Return the expires dates sorted desc
+        let expiryDateValues = receiptsInfo
+            .map { (receipt) -> NSString? in
+                let key = duration != nil ? "original_purchase_date_ms" : "expires_date_ms"
+                return receipt[key] as? NSString
+            }
+            .filter { (dateString) -> Bool in
+                return dateString != nil
+            }
+            .map { (dateString) -> NSDate in
+                // If duration is set, create an "expires date" value calculated from the original purchase date
+                let addDuration = duration ?? 0
+                let expiresDateDouble = (dateString!.doubleValue / 1000 + addDuration)
+                return NSDate(timeIntervalSince1970: expiresDateDouble)
+            }
+            .sort { (a, b) -> Bool in
+                // Sort by descending date order
+                return a.compare(b) == .OrderedDescending
+            }
+      
+        guard let firstExpiryDate = expiryDateValues.first else {
+            return .NotPurchased
+        }
```

Comment:
As we sort in the step above, the `firstExpiryDate` is the farthest in the future.